python/semantic_kernel/connectors/memory/azure_cosmos_db/utils.py

```diff
@@ -0,0 +1,227 @@
+
+    Args:
+        data_model_definition (VectorStoreRecordDefinition): The definition of the data model.
+
+    Returns:
+        dict[str, Any]: The vector embedding policy.
+    """
+    vector_embedding_policy: dict[str, Any] = {"vectorEmbeddings": []}
+
+    for _, field in data_model_definition.fields.items():
+        if isinstance(field, VectorStoreRecordVectorField):
+            vector_embedding_policy["vectorEmbeddings"].append({
+                "path": f'/"{field.name}"',
+                "dataType": to_datatype(field.property_type),
+                "distanceFunction": to_distance_function(field.distance_function),
+                "dimensions": field.dimensions,
+            })
+
+    return vector_embedding_policy
+
+
+def get_key(key: str | AzureCosmosDBNoSQLCompositeKey) -> str:
+    """Gets the key value from the key.
+
+    Args:
+        key (str | AzureCosmosDBNoSQLCompositeKey): The key.
+
+    Returns:
+        str: The key.
+    """
+    if isinstance(key, AzureCosmosDBNoSQLCompositeKey):
+        return key.key
+
+    return key
+
+
+def get_partition_key(key: str | AzureCosmosDBNoSQLCompositeKey) -> str:
+    """Gets the partition key value from the key.
+
+    Args:
+        key (str | AzureCosmosDBNoSQLCompositeKey): The key.
+
+    Returns:
+        str: The partition key.
+    """
+    if isinstance(key, AzureCosmosDBNoSQLCompositeKey):
+        return key.partition_key
+
+    return key
+
+
+def build_query_parameters(
+    data_model_definition: VectorStoreRecordDefinition,
+    keys: Sequence[str | AzureCosmosDBNoSQLCompositeKey],
+    include_vectors: bool,
+) -> tuple[str, list[dict[str, Any]]]:
+    """Builds the query and parameters for the Azure Cosmos DB NoSQL query item operation.
+
+    Args:
+        data_model_definition (VectorStoreRecordDefinition): The definition of the data model.
+        keys (Sequence[str | AzureCosmosDBNoSQLCompositeKey]): The keys.
+        include_vectors (bool): Whether to include the vectors in the query.
+
+    Returns:
+        tuple[str, list[dict[str, str]]]: The query and parameters.
+    """
+    included_fields = [
+        field
+        for field in data_model_definition.field_names
+        if include_vectors or field not in data_model_definition.vector_field_names
+    ]
+    if data_model_definition.key_field_name != COSMOS_ITEM_ID_PROPERTY_NAME:
+        # Replace the key field name with the Cosmos item id property name
+        included_fields = [
+            field if field != data_model_definition.key_field_name else COSMOS_ITEM_ID_PROPERTY_NAME
+            for field in included_fields
+        ]
+
+    select_clause = ", ".join(f"c.{field}" for field in included_fields)
+
+    return (
+        f"SELECT {select_clause} FROM c WHERE c.id IN ({', '.join([f'@id{i}' for i in range(len(keys))])})",  # nosec: B608
+        [{"name": f"@id{i}", "value": get_key(key)} for i, key in enumerate(keys)],
+    )
+
+
+class CosmosClientWrapper(CosmosClient):
```

Comment:
@eavanvalkenburg With the newly introduce async context manager pattern, would it create issue when a client is closed twice?

Reply:
People can still use the connector without the async context manager. So I guess we need to keep it.